hydra-node/src/Hydra/API/ServerOutput.hs

```diff
@@ -161,15 +163,46 @@ prepareServerOutput OutputCBOR response =
         Fanout -> encodedResponse
     TxValid{Hydra.API.ServerOutput.transaction = tx} -> replacedResponse tx
     TxInvalid{Hydra.API.ServerOutput.transaction = tx} -> replacedResponse tx
-    SnapshotConfirmed{} -> encodedResponse
+    SnapshotConfirmed{Hydra.API.ServerOutput.snapshot} ->
+      replacedSnapshotTxResponse (confirmed snapshot)
     GetUTxOResponse{} -> encodedResponse
     InvalidInput{} -> encodedResponse
     Greetings{} -> encodedResponse
-    PostTxOnChainFailed{} -> encodedResponse
+    PostTxOnChainFailed{postChainTx} ->
+      case postChainTx of
+        CloseTx{confirmedSnapshot} ->
+          case confirmedSnapshot of
+            InitialSnapshot{} -> encodedResponse
+            ConfirmedSnapshot{Hydra.Snapshot.snapshot} ->
+              replacedSnapshotTxInPostTxOnChainFailedResponse (confirmed snapshot)
+        ContestTx{confirmedSnapshot} ->
+          case confirmedSnapshot of
+            InitialSnapshot{} -> encodedResponse
+            ConfirmedSnapshot{Hydra.Snapshot.snapshot} ->
+              replacedSnapshotTxInPostTxOnChainFailedResponse (confirmed snapshot)
```

Comment:
Coudld: inline to avoid long names

```suggestion
              encodedResponse &
          key "postChainTx" .
          key "confirmedSnapshot" .
          key "snapshot" .
          key "confirmedTransactions" .~ toJSON $ toCborHex <$> confirmed snapshot
```

assuming there is a `toCborHex :: Tx -> Text`